### ISSUE TITLE
p-table: RowGroup not grouping on subsequent pages

### ISSUE THREAD

**Issue by psarno** [primefaces/primeng#11764]
### Describe the bug

We have a p-table component using `[lazy]="true"` and the following ng-templates:

`<ng-template pTemplate="groupheader" let-rowData let-rowIndex="rowIndex" let-expanded="expanded">`

`<ng-template pTemplate="rowexpansion" let-rowData let-columns="columns">`

We also have the row toggle as `[pRowToggler]="rowData"`.

On the first page, it correctly groups by the `groupheader` / `pRowToggler`, but on all subsequent pages the grouping stops and each line is shown individually.

This was working prior to upgrading to PrimeNG 14.

Page 1:

![image](https://user-images.githubusercontent.com/18640860/182227763-eb46c9d8-3041-4c02-a2db-400a93dc88dd.png)

Page 2:

![image](https://user-images.githubusercontent.com/18640860/182227977-1929fc45-b217-46c2-a034-a00b8b7ee3e0.png)

### Environment

Angular 14.0.3 / PrimeNG 14.0.0 testing with Chrome.

### Angular version

14.0.3

### PrimeNG version

14.0.0

### Build / Runtime

TypeScript

### Language

TypeScript

### Node version (for AoT issues node --version)

16.13.1

### Steps to reproduce the behavior

1. Lazy load data into a p-table component with a paginator
2. Click on any page other than page 1
3. Note that the grouping is no longer applied

### Expected behavior

p-table continues to group by the unique `pRowToggler` as defined in the `groupheader`, as it does on page 1.

**Comment by cetincakiroglu** [comment]
Hi, 

Cannot replicate the issue please feel free to re-open with a StackBlitz example.

**Comment by psarno** [comment]
@cetincakiroglu I have created a repro for this in Stackblitz.  Please note on the second page it no longer groups as it does on page 1.

Repro Stackblitz:

https://stackblitz.com/edit/github-sb1bvu?file=src/app/app.component.ts

![image](https://user-images.githubusercontent.com/18640860/183762541-3f9d5f73-185f-4b79-a93c-613daa7500a9.png)

**Comment by domagoj-telum** [comment]
I am having the same issue. @psarno did you manage to solve this or find what the problem was?

**Comment by psarno** [comment]
@domagoj-telum No, and I have included a reproducer, I'm not sure why this is closed as unable to reproduce.

**Comment by psarno** [comment]
@cetincakiroglu This was closed as can not reproduce but I've provided a reproducer that shows the issue.

**Comment by cetincakiroglu** [comment]
@psarno Re-opened again 👍🏻

**Comment by winglam209** [comment]
Hi, any updates on this…? Seems like this issue still persists on primeng v15 and v17…

**Comment by alexandru1124** [comment]
did someone manage to make a workaround for this?
